core/services/ocr2/plugins/ocr2keeper/evm21/logprovider/provider.go

```diff
@@ -408,7 +408,9 @@ func (p *logEventProvider) readLogs(ctx context.Context, latest int64, filters [
 
 		p.buffer.enqueue(filter.upkeepID, filteredLogs...)
 
-		filter.lastPollBlock = latest
+		// Update the lastPollBlock for filter in slice this is then
+		// updated into filter store in updateFiltersLastPoll
+		filters[i].lastPollBlock = latest
```

Comment:
I wish we used more pointers so this change wouldn't be necessary lol

Reply:
yeah i personally find this very confusing, seems like slices are passed by reference by default